app/assets/javascripts/uploads.js

```diff
@@ -26,6 +27,27 @@
     }
   }
 
+  Danbooru.Upload.initialize_submit = function() {
+    $("#submit-button").click(function(e) {
```

Comment:
This won't trigger if you submit the upload by pressing the enter key instead of clicking the submit button.

This could be fixed by binding to the form submit event instead (`$("#form").submit(function (e) ...)`) then saying `e.preventDefault()` in the error branch (prevent the submission from going through on error, rather than submitting it on success).